packages/salesforcedx-vscode-core/src/util/metadataDictionary.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { LocalComponent } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { PathStrategyFactory, SourcePathStrategy } from '../commands/util';
+
+const DEFINITIONS: { [key: string]: MetadataInfo } = {
+  apexclass: {
```

Comment:
Why do we need the definitions for ApexClass, ApexTrigger, etc if not having a definition defaults to using `PathStrategyFactory.createDefaultStrategy()` in `OverwriteComponentPrompt.componentExists` ?

Reply:
The definitions aren't just for the path strategy, but to hold any meta information that's needed in the extensions. The template commands will be able to reference the extensions/directory information for these types.